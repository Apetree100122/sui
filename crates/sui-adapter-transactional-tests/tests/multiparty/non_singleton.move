// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

//# init --addresses ex=0x0

//# publish
module ex::m;

public struct Pub has key, store {
    id: UID,
}

public struct Priv has key {
    id: UID,
}

public fun mint(ctx: &mut TxContext) {
    let p = Pub { id: object::new(ctx) };
    let q = Priv { id: object::new(ctx) };
    transfer::public_transfer(p, ctx.sender());
    transfer::transfer(q, ctx.sender());
}

public fun priv_multiparty(obj: Priv, p: vector<address>) {
    transfer::multiparty_transfer(obj, p)
}

//# run ex::m::mint

// Aborts in transfer::multiparty_transfer since the party is empty
//# run ex::m::priv_multiparty --args object(2,0) vector[]

// Aborts in transfer::multiparty_transfer since the party is not length 1
//# run ex::m::priv_multiparty --args object(2,0) vector[@0,@1]

// Aborts in transfer::public_multiparty_transfer since the party is empty
//# run sui::transfer::public_multiparty_transfer
//#     --type-args ex::m::Pub
//#     --args object(2,1) vector[]

// Aborts in transfer::public_multiparty_transfer since the party is not length 1
//# run sui::transfer::public_multiparty_transfer
//#     --type-args ex::m::Pub
//#     --args object(2,1) vector[@0,@1]
