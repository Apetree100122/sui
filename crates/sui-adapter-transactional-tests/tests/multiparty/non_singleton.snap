processed 7 tasks

task 1, lines 6-26:
//# publish
created: object(1,0)
mutated: object(0,0)
gas summary: computation_cost: 1000000, storage_cost: 6353600,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, lines 28-30:
//# run ex::m::mint
created: object(2,0), object(2,1)
mutated: object(0,0)
gas summary: computation_cost: 1000000, storage_cost: 3442800,  storage_rebate: 978120, non_refundable_storage_fee: 9880

task 3, lines 31-33:
//# run ex::m::priv_multiparty --args object(2,0) vector[]
Error: Transaction Effects Status: Move Runtime Abort. Location: sui::transfer::multiparty_transfer (function index 2) at offset 7, Abort Code: 9223372350387716102
Execution Error: ExecutionError: ExecutionError { inner: ExecutionErrorInner { kind: MoveAbort(MoveLocation { module: ModuleId { address: sui, name: Identifier("transfer") }, function: 2, instruction: 7, function_name: Some("multiparty_transfer") }, 9223372350387716102), source: Some(VMError { major_status: ABORTED, sub_status: Some(9223372350387716102), message: Some("sui::transfer::multiparty_transfer at offset 7"), exec_state: None, location: Module(ModuleId { address: sui, name: Identifier("transfer") }), indices: [], offsets: [(FunctionDefinitionIndex(2), 7)] }), command: Some(0) } }

task 4, lines 34-36:
//# run ex::m::priv_multiparty --args object(2,0) vector[@0,@1]
Error: Transaction Effects Status: Move Runtime Abort. Location: sui::transfer::multiparty_transfer (function index 2) at offset 7, Abort Code: 9223372350387716102
Execution Error: ExecutionError: ExecutionError { inner: ExecutionErrorInner { kind: MoveAbort(MoveLocation { module: ModuleId { address: sui, name: Identifier("transfer") }, function: 2, instruction: 7, function_name: Some("multiparty_transfer") }, 9223372350387716102), source: Some(VMError { major_status: ABORTED, sub_status: Some(9223372350387716102), message: Some("sui::transfer::multiparty_transfer at offset 7"), exec_state: None, location: Module(ModuleId { address: sui, name: Identifier("transfer") }), indices: [], offsets: [(FunctionDefinitionIndex(2), 7)] }), command: Some(0) } }

task 5, lines 37-41:
//# run sui::transfer::public_multiparty_transfer --type-args ex::m::Pub --args object(2,1) vector[]
Error: Transaction Effects Status: Move Runtime Abort. Location: sui::transfer::public_multiparty_transfer (function index 3) at offset 7, Abort Code: 9223372397632356358
Execution Error: ExecutionError: ExecutionError { inner: ExecutionErrorInner { kind: MoveAbort(MoveLocation { module: ModuleId { address: sui, name: Identifier("transfer") }, function: 3, instruction: 7, function_name: Some("public_multiparty_transfer") }, 9223372397632356358), source: Some(VMError { major_status: ABORTED, sub_status: Some(9223372397632356358), message: Some("sui::transfer::public_multiparty_transfer at offset 7"), exec_state: None, location: Module(ModuleId { address: sui, name: Identifier("transfer") }), indices: [], offsets: [(FunctionDefinitionIndex(3), 7)] }), command: Some(0) } }

task 6, lines 42-44:
//# run sui::transfer::public_multiparty_transfer --type-args ex::m::Pub --args object(2,1) vector[@0,@1]
Error: Transaction Effects Status: Move Runtime Abort. Location: sui::transfer::public_multiparty_transfer (function index 3) at offset 7, Abort Code: 9223372397632356358
Execution Error: ExecutionError: ExecutionError { inner: ExecutionErrorInner { kind: MoveAbort(MoveLocation { module: ModuleId { address: sui, name: Identifier("transfer") }, function: 3, instruction: 7, function_name: Some("public_multiparty_transfer") }, 9223372397632356358), source: Some(VMError { major_status: ABORTED, sub_status: Some(9223372397632356358), message: Some("sui::transfer::public_multiparty_transfer at offset 7"), exec_state: None, location: Module(ModuleId { address: sui, name: Identifier("transfer") }), indices: [], offsets: [(FunctionDefinitionIndex(3), 7)] }), command: Some(0) } }
